Modules/_pickle.c

```diff
@@ -2184,8 +2184,7 @@ _Pickler_write_bytes(PicklerObject *self,
         /* Stream write the payload into the file without going through the
            output buffer. */
         if (payload == NULL) {
-            payload = mem = PyMemoryView_FromMemory((char *) data, data_size,
-                                                    PyBUF_READ);
+            payload = mem = PyBytes_FromStringAndSize(data, data_size);
```

Comment:
This forces a memory copy. Wouldn't it be possible to create a read-only memoryview that keeps a reference to the original ascii `str` object to avoid the large memory overhead?

Reply:
This is not easy. Currently there is no API for creating a memoryview to the area of memory with linked Python object. Designing such API is a separate issue. There should be other issues on the tracker that needs this API. For example bytes IO seems suffers from this issue.

Here I just try to fix a regression introduced by #4353.

Reply:
I agree.